ServiceLocator/Packages/com.virtualmaker.servicelocator/Runtime/ServiceLocator.cs

```diff
@@ -50,31 +57,68 @@ private static ServiceLocator GetOrCreate()
         }
 
         public static T Get<T>() where T : IService
+        {
+            if (TryGet(out T service))
+            {
+                return service;
+            }
+
+            if (_instance == null)
+            {
+                throw new Exception($"Service Locator not found");
+            }
+
+            throw new Exception($"Service {typeof(T)} not found");
+        }
+
+        public static bool TryGet<T>(out T service) where T : IService
         {
             var instance = GetOrCreate();
 
             if (instance == null)
             {
-                throw new Exception("ServiceLocator not found");
+                // instance can be null if called while stopping play mode
+                // because ServiceLocator is destroyed before other objects.
+                service = default!;
+                return false;
             }
 
-            if (!instance._services.TryGetValue(typeof(T), out var service))
+            if (!instance._services.TryGetValue(typeof(T), out var svc))
             {
                 // Check if the service was just added in editor.
                 if (!Application.isPlaying)
                 {
                     instance.RegisterServices();
 
-                    if (instance._services.TryGetValue(typeof(T), out service))
+                    if (instance._services.TryGetValue(typeof(T), out svc))
                     {
-                        return (T)service;
+                        service = (T)svc;
+                        return true;
                     }
                 }
 
-                throw new Exception($"Service {typeof(T)} not found");
+                service = default!;
+                return false;
             }
 
-            return (T)service;
+            service = (T)svc;
+            return true;
+        }
+
+        private void OnEnable()
+        {
+            foreach (var service in _servicesInOrder)
+            {
+                service.Initialize();
```

Comment:
kinda wish these had the same name as OnEnable/OnDisable so they're more obvious about where in the lifecycle of events they are called form.